pkg/client/client.go

```diff
@@ -315,8 +315,14 @@ type imageFactory struct {
 	keychain     authn.Keychain
 }
 
-func (f *imageFactory) NewImage(repoName string, daemon bool, imageOS string) (imgutil.Image, error) {
-	platform := imgutil.Platform{OS: imageOS}
+func (f *imageFactory) NewImage(repoName string, daemon bool, target dist.Target) (imgutil.Image, error) {
+	platform := imgutil.Platform{OS: target.OS, Architecture: target.Arch}
+
+	platform.Variant = target.ArchVariant
+	if len(target.Distributions) > 0 {
+		// We assume the given target's distributions were already expanded, we should be dealing with just 1 distribution name and version.
```

Comment:
```suggestion
		// We need to set platform distribution information so that it will be reflected in the image config.
		// We assume the given target's distributions were already expanded; we should be dealing with just 1 distribution name and version.
```